src/drivers/stm32/drv_io_timer.c

```diff
@@ -374,16 +378,45 @@ static int allocate_channel(unsigned channel, io_timer_channel_mode_t mode)
 
 static int timer_set_rate(unsigned timer, unsigned rate)
 {
-
-	/* configure the timer to update at the desired rate */
-	rARR(timer) = 1000000 / rate;
+	/* Configure the timer to run at 1MHz (other system code _assumes_ that timers run at 1MHz).
+	 * If you want to hack this to achieve a different frequency, alter the value of
+	 * .clock_freq in the relevant driver/boards/xxxx/xxx_timer_config.c file.
+	 */
+	timer_freq[timer] = 1;
+	rPSC(timer) = (io_timers[timer].clock_freq / 1000000) - 1;
```

Comment:
How does changing the prescaler affect the output pins on a rate change? I am just noting this because it is a different sequencing then the original code. It only set the prescaler once at init time not an a rate changes.